### PR TITLE
Forbid registering invalid file inputs via the runtime API

### LINKED ISSUE #6280: Forbid using TaskInputs#file with something that isn't a file
https://github.com/gradle/gradle/blob/54e42e81171fd5e4b738b0401829cd839dae74f9/subprojects/core/src/main/java/org/gradle/api/internal/tasks/DefaultTaskInputs.java#L224

### LINKED ISSUE #6282: Fail when registering invalid inputs and outputs via TaskInputs and TaskOutputs methods
https://github.com/gradle/gradle/blob/54e42e81171fd5e4b738b0401829cd839dae74f9/subprojects/core/src/main/java/org/gradle/api/internal/tasks/TaskValidationContext.java#L30

### PR DESCRIPTION
**Pull request by wolfs**
Fixes #6280.
Fixes #6282.

### CODE REVIEW

File: subprojects/core/src/test/groovy/org/gradle/api/internal/changedetection/changes/DefaultTaskArtifactStateRepositoryTest.groovy [hunk 1]
```diff
@@ -816,7 +816,9 @@ class DefaultTaskArtifactStateRepositoryTest extends AbstractProjectBuilderSpec
                 }
             }
             if (inputProperties != null) {
-                task.getInputs().properties(inputProperties)
+                inputProperties.each { key, value ->
+                    task.getInputs().property(key, value).optional(true)
+                }
```
Comment (marcphilipp) [inline subprojects/core/src/test/groovy/org/gradle/api/internal/changedetection/changes/DefaultTaskArtifactStateRepositoryTest.groovy]: Why was this change necessary?

Comment (wolfs) [reply subprojects/core/src/test/groovy/org/gradle/api/internal/changedetection/changes/DefaultTaskArtifactStateRepositoryTest.groovy]: One of the tests registers `null` as an input via the runtime API. This causes an exception now.

File: subprojects/docs/src/docs/release/notes.md [hunk 2]
```diff
@@ -94,6 +94,7 @@ The previously deprecated support for Play Framework 2.2 has been removed.
 - Removed the property `styleSheet` from `ScalaDocOptions`.
 - Forbid passing `null` as configuration action to the methods `from` and `to` on `CopySpec`.
 - Removed the property `bootClasspath` from `CompileOptions`.
+- Registering invalid inputs or outputs via the runtime API is now an error.
```
Comment (lptr) [inline subprojects/docs/src/docs/release/notes.md]: Should we explain what "invalid" means?

Comment (lptr) [reply subprojects/docs/src/docs/release/notes.md]: Also, do we need to change anything in the user guide about this?

**Pull request merged**